lib/va_profile/prefill/military_information.rb

```diff
@@ -321,7 +321,11 @@ def guard_reserve_service_by_date
         end
 
         all_episodes.sort_by do |episode|
-          episode.end_date || (Time.zone.today + 3650)
+          if episode.end_date.blank? # Handles nil and empty string
+            Time.zone.today + 3650
+          else
+            Date.parse(episode.end_date)
+          end
```

Comment:
A couple questions:
1. Code coverage shows both of these conditions getting hit?
2. How did you decide on `Date.parse` here?

Reply:
I decided on Date.parse because the value is a string on the model/object. Service episodes by date uses the same: https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/va_profile/prefill/military_information.rb#L251

Reply:
got it. thanks!